corehq/apps/cleanup/management/commands/evaluate_couch_model_for_sql.py

```diff
@@ -13,7 +13,14 @@
 class Command(BaseCommand):
     """
         Given a couch document type, iterates over all documents and reports back
-        on whether those attributes are ever null and what their largest values are.
+        on usage of each attribute, to aid in selecting SQL fields for those attributes.
+
+        For each attribute report:
+        - Whether the value is ever null or empty, for the purpose of deciding whether to use null=True
```

Comment:
I'm curious why you would decide to use `null=True` if there are no null values but there are empty (string) values? Typically it's better to use `null=False` if possible because null values make all kinds of things harder and/or more error prone.

Reply:
I hadn't considered that. So is it better to use `null=False, blank=True` if there are falsy values but no `None` values, and then only use `null=True` (which I'm guessing implies `blank=True`) if there are `None` values?

Reply:
`null=True` is always preferred if possible. It adds a `NOT NULL` constraint on the database column.

`blank` is a form validation flag that has no effect on the database (it should be called `required`, and have the opposite meaning). I think you'll need to look at if and how Django form validation is used in association with the model to know the correct value of this flag.

See also
- https://docs.djangoproject.com/en/dev/ref/models/fields/#field-options
- https://stackoverflow.com/a/47590145/10840

Reply:
Cool, thank you. I'm likely to update this script again soon, and will then update the logic around empty values.

Reply:
5903dc1bbeac4006d5329f93ac7036541634eea2